source/dotnet/Library/AdaptiveCards/AdaptiveHeight.cs

```diff
@@ -51,6 +51,38 @@ public AdaptiveHeight()
         {
         }
 
+        /// <summary>
+        /// Initializes an AdaptiveHeight instance with the given string "auto" |"stretch" or "100px"
+        /// </summary>
+        /// <param name="value">enumeration value or pixel .</param>
+        public AdaptiveHeight(string value)
+        {
+            if (value == null)
+            {
+                HeightType = AdaptiveHeightType.Auto;
+            }
+            else if (value.EndsWith("px"))
+            {
+                value = value.Substring(0, value.Length - 2);
+                HeightType = AdaptiveHeightType.Pixel;
+                Unit = uint.Parse(value);
+            }
+            else if (uint.TryParse(value, out var val))
+            {
+                HeightType = AdaptiveHeightType.Pixel;
+                Unit = val;
+            }
+            else
+            {
+                HeightType = (AdaptiveHeightType)Enum.Parse(typeof(AdaptiveHeightType), value, ignoreCase: true);
```

Comment:
`Enum.TryParse` and gracefully fall back to `Auto` (this is what JS renderer does)?

same for `AdaptiveWidth`...

Reply:
We want to throw the exception so the converter can add warning text. Added a comment to this effect.